src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -553,9 +553,6 @@ private void finishImportList() {
             final String importGroup = importObject.getImportGroup();
             final String fullImportIdent = importObject.getImportFullPath();
 
-            if (getCountOfEmptyLinesBefore(importObject.getLineNumber()) > 1) {
```

Comment:
After spending a lot of time with this PR....

I do not like this change and I recall my approval for extra regression.
Issue is simple, and fix should also be simple.
Please change loop that iterates over imports, and skip reporting of extra line before first import.

There should be no regression at all, only required case should be fixed.

Reply:
Personally, I dislike the change introduced by https://github.com/checkstyle/checkstyle/commit/914dd9994c9f108f54849a96bc35bbdd85de1053 . For it combines the checks of `separateLineBetweenGroups` and separateLineWithinGroup. A clear implementation should be simple and straight.
If I have ever had a choice, I'd like to completely reimplementation #3551 rather than  hacking.

> Issue is simple, and fix should also be simple.

I agree, but why not make it more maintainable, since there's another issue introduced by https://github.com/checkstyle/checkstyle/commit/914dd9994c9f108f54849a96bc35bbdd85de1053

Before this PR, I had another solution for #3551, just a few lines, much less than this. I'll create another PR and leave this one unchanged.

Reply:
Ok, even better :), I hope it will not make any regression